src/libs/ReportUtils.js

```diff
@@ -1882,6 +1967,47 @@ function buildOptimisticReportPreview(chatReport, iouReport, comment = '') {
     };
 }
 
+/**
+ * Builds an optimistic modified expense action with a randomly generated reportActionID.
+ *
+ * @param {Object} transactionThread
+ * @param {Object} oldTransaction
+ * @param {Object} transactionChanges
+ * @param {Object} isFromExpenseReport
+ * @returns {Object}
+ */
+function buildOptimisticModifiedExpenseReportAction(transactionThread, oldTransaction, transactionChanges, isFromExpenseReport) {
+    const originalMessage = getModifiedExpenseOriginalMessage(oldTransaction, transactionChanges, isFromExpenseReport);
+    return {
+        actionName: CONST.REPORT.ACTIONS.TYPE.MODIFIEDEXPENSE,
+        actorAccountID: currentUserAccountID,
```

Comment:
[This issue](https://github.com/Expensify/App/issues/26392) is probably caused by the change introduce in this PR.

Issue: Changing Currency in IOU Causes Account Email to Replace Request Email
Steps to reproduce: 

- Click on the plus sign and select "Request Money".
- Enter the desired amount and email address for the money request.
- Double-click on the requested money in IOU and change the currency by clicking on the amount.
- Go back twice and observe that the requested email has been replaced by the account email.